File: backend/app/schemas/thread.py
from datetime import datetime

from app.models import thread as model_thread
from app.schemas import base as schema_base
from app.schemas import user as schema_user


class Thread(schema_base.ORMBaseModel):
    rid: int
    depth: int
    user: schema_user.User
    state: model_thread.TypeThreadState
    note: str
    created_at: datetime


class ThreadReport(schema_base.ORMBaseModel):
    rid_projects: int
    threads: list[Thread]


class ThreadStatus(schema_base.ORMBaseModel):
    rid_projects: int
    count: int
    is_important: bool


class ThreadCreate(schema_base.ORMBaseModel):
    rid_projects: int
    rid_parent: int | None = None
    state: model_thread.TypeThreadState
    note: str


class ThreadUpdate(schema_base.ORMBaseModel):
    rid: int
    state: model_thread.TypeThreadState
    note: str

File: frontend/app/src/types/Thread.ts
import type { User } from '@/types/User'

// prettier-ignore
export enum TypeThreadState {
    NONE      = 0,
    RUN       = 1,
    IMPORTANT = 2,
    COMPLETED = 3,
  }

export interface Thread {
  rid: number
  depth: number
  user: User
  state: TypeThreadState
  note: string
  created_at: string
}

export interface ThreadStatus {
  rid_projects: number
  count: number
  is_updated: boolean
}

export interface ThreadCreate {
  rid_projects: number
  rid_parent: number | null
  state: TypeThreadState
  note: string
}

export interface ThreadUpdate {
  rid: number
  rid_projects: number
  state: TypeThreadState
  note: string
}
